### ISSUE TITLE
Darker shades?

### ISSUE THREAD

**Issue by brentluna** [dracula/visual-studio-code#30]
Any chance of adding a 2nd shade option w/ slightly flatter colors? I love the combination, of colors, but can be a bit too bright on the eyes for me personally. Just a suggestion, understand if you're not interested.

**Comment by dsifford** [comment]
@brentluna Yep, I'm open to that. How about collaborating with @Eric-Jackson on #29?

**Comment by dsifford** [comment]
Are you talking about the _entire_ theme  or just parts of the UI? The theme is locked to the Dracula scheme. UI is a bit more flexible.

If you're looking for a darker theme, I beleve there's a few forks available in the marketplace.

**Comment by matt3224** [comment]
@dsifford Here's a quick comparison with the dracula coda theme which is much easier on the eyes and has better highlighting imho

![screen shot 2017-05-07 at 3 03 27 pm](https://cloud.githubusercontent.com/assets/2401925/25782037/5b4517e2-333a-11e7-8805-4469faaaf76c.png)

**Comment by dsifford** [comment]
@matt3224 Coda is doing something behind the scenes to soften the color of that theme. I checked the [theme repo](https://github.com/dracula/coda) and it is definitely using the same color that the rest of the official dracula themes are using for the background color.

Unfortunately, there's nothing I can do to fix that in this repo because I'm bound to the Dracula spec.

You have two options though:

1. Give this one a try: https://marketplace.visualstudio.com/items?itemName=yomed.theme-dracula-soft
2. Change this theme's background color in your User Settings: 

```json
"workbench.colorCustomizations": {
     "editor.background": "<your-color-here>"
}
```

**Comment by brentluna** [comment]
@dsifford I'm not familiar w/ theming  code, but noticed that it basically had key value type system  w/ hex colors, my basic thought was taking those  hex colors and basically finding the next shade darker to bring down the neon factor just a bit. I know some themes come packaged w/ multiple versions, such as dark/light or different shades, so wasn't sure if that was possible w/ dracula.   

The dracula soft is nice, but figured it doesn't look like it's going to be updated w/ the official as it's last been touched 10/2016.

**Comment by dsifford** [comment]
>  I know some themes come packaged w/ multiple versions, such as dark/light or different shades, so wasn't sure if that was possible w/ dracula.

Can you give me an example of one of these themes? I'm not familar with how to do that.

**Comment by Eric-Jackson** [comment]
https://marketplace.visualstudio.com/items?itemName=tomphilbin.gruvbox-themes

**Comment by dsifford** [comment]
Thanks @Eric-Jackson! Super helpful.

I'll check this out later this evening.

**Comment by dsifford** [comment]
Alright I think I got it figured out.

For now, I think we could start by including the standard theme and a single "Dracula Soft" variant.

Here's a comparison of the two:

![comparison-gif](https://media.giphy.com/media/3og0ItXb4s5yE5aQxO/giphy.gif)

### Higher resolution comparison

**Standard:**

![standard](https://cloud.githubusercontent.com/assets/5240018/25831693/d98db7e0-3433-11e7-8839-213143385382.png)

**Soft:**

![soft](https://cloud.githubusercontent.com/assets/5240018/25831665/af92e6c2-3433-11e7-92ca-04be5569d5e1.png)

In short, the soft theme is identical, but brightened (not to be confused with lightened) by 3%, desaturated by 8%, and then darkened by 3%. I know that probably sounds like a weird combination of changes, but that was the best one I think after trying a handful of combinations.

Let me know what you all think.

**Comment by dsifford** [comment]
Actually, I think this might be better:

Brightened by 8%, desaturated by 3%, darkened by 5%

![image](https://cloud.githubusercontent.com/assets/5240018/25831808/8347395a-3434-11e7-9635-1da1fcb8a75d.png)

**Comment by dsifford** [comment]
Any thoughts on this @brentluna @matt3224?

**Comment by matt3224** [comment]
@dsifford Yes! These look good, I think you could retain the original background colour though maybe, it seems yellowy.

Would be interesting to test it and see if my eyes approve after a coding session!

**Comment by brentluna** [comment]
@dsifford Awesome, I think it looks like a good start! My first thought would be to not keep the original background, but that is easily changeable enough w/ VS's new theme settings, so not a huge deal on that specifically. 

Looking forward to trying it out.

**Comment by matt3224** [comment]
@brentluna hmmm actually, I have set mine to a custom shade already so by "retain the background colour" what I mean is.. try this `#303241` 🙃

**Comment by dsifford** [comment]
@brentluna @matt3224 How about we just try the one I mentioned above as a starting point. I'd like the stay on the path of only messing with the saturation, brightness, and lightness so that the hues stay where they're supposed to be and so that everything adjusts in unison.

After a day or two of kicking the tires we can revisit and see what adjustments can be made from there. Sound fair?

@matt3224 To comment on your remark about how it appears "yellowy". I think that might just be the screenshot. On my screen it looks more "grayey" than "yellowy" 😝

**Comment by matt3224** [comment]
Ah ok, no worries. Do you need to commit or are you distributing in some other way?

**Comment by brentluna** [comment]
@dsifford I agree, I like the general color scheme, just was hoping to tone down the neon factor a bit. Looking forward to you pushing the commit

**Comment by dsifford** [comment]
Haven't heard back on this so I'm gonna assume we're all good and you both are satisfied with the current `Dracula Soft` implementation.

Feel free to reopen this if not.

**Comment by matt3224** [comment]
Hey @dsifford sorry, hectic week.

I've used it all week and do feel that there is room for improvement. I feel that the background is too black, i feel it needs to be more blue and a touch lighter, second to that the text colours are still too harsh (i literally stare at this all day).

**Comment by brentluna** [comment]
I do agree on the actual syntax color, that it in my opinion it seems to have been too small of a chamge, especially with the green that is still extremely bright and seems to often be the primary color.

**Comment by dsifford** [comment]
@brentluna  @matt3224 Feel free to send a PR, keeping in mind that I'm still pretty firm on keeping the hues where they're supposed to be.

**Comment by dsifford** [comment]
Reopening this from comments received in #45

**Comment by jaredreich** [comment]
I think Dracula-Soft could use more desaturation. On my macbook retina screen it can be very hard on the eyes at higher screen brightness levels. I think the whole point of Dracula-Soft is to give people the same color scheme but keep it easier on the eyes, so maybe a bit more desaturation is all it needs :)

**Comment by dsifford** [comment]
@jaredreich (et al)

Finally got a few minutes to sit down with a mac and hammer out a better soft implementation. I ended up just heavily desaturating the brighter colors and leaving the darker/softer colors (backgrounds, whites, etc) as-is.

Gonna push out a quick update with just those changes here in a few. Let me know what you think.

I'll leave this issue open until I hear back.

**Comment by jaredreich** [comment]
@dsifford Awesome, so great to hear, thanks! I see the published release here on GH, but I'm not sure how to test this locally, will the update be on the official extension?

**Comment by dsifford** [comment]
Yep, I pushed out the update to the extension marketplace about 5 minutes ago. Should show up for you shortly. 👍

**Comment by jaredreich** [comment]
@dsifford The update just showed up for me in VSCode... Awesome work! The changes are perfect, much easier on the eyes while keeping the color palette. Hope this works out for everyone else using the theme.

**Comment by ajitid** [comment]
> Notes: Mac users should have a better experience with this one.

I'm on Windows and I'm not able to find even a single difference between Dracula and Dracula soft:

![screenshot 166](https://user-images.githubusercontent.com/26769888/27563204-efe60e1c-5aed-11e7-8ce4-6dee609aa124.png)

![screenshot 167](https://user-images.githubusercontent.com/26769888/27563190-ddbe521c-5aed-11e7-999b-24bf3eec8a04.png)

**Comment by ajitid** [comment]
> Note: Mac users should have a better experience with this one.

I'm on Windows and I am not able to find any difference between Dracula and Dracula Soft

![screenshot 166](https://user-images.githubusercontent.com/26769888/27563372-daa6a466-5aee-11e7-9b6a-e4d0d23ed1f5.png)

![screenshot 167](https://user-images.githubusercontent.com/26769888/27563389-f1bbd55e-5aee-11e7-999f-979b0e285f5c.png)

**Comment by ajitid** [comment]
> Mac users should have a better experience with this one.

I'm on Windows and I'm not able to find any difference between Dracula and Dracula Soft

![screenshot 166](https://user-images.githubusercontent.com/26769888/27563472-7834f35e-5aef-11e7-9a2c-b1a4c71f9d8a.png)

![screenshot 167](https://user-images.githubusercontent.com/26769888/27563473-7839be20-5aef-11e7-9cd5-b2288aef5dec.png)

**Comment by jaredreich** [comment]
@ajitid I can tell the slight desaturation difference between your two screenshots. Must be the screen you're using.

**Comment by ajitid** [comment]
I too saw that. Can you open each screenshot and check it again?

**Comment by dsifford** [comment]
@ajitid That's expected. Macbooks or iMac displays make the brighter colors of the theme much more intense and hard on the eyes. The goal of the soft variant was to target that issue specifically. 

I'm really not wanting to stray too far from the standard Dracula theme otherwise. Having said that, if more people speak up about it, I'll consider a happy medium between the old soft theme and the new one.

**Comment by iamandreymarchenko** [comment]
@dsifford thanks for your work, mate. But i want to say that on my macbook pro i can't see any difference between original and soft version. It would be awesome if in a future we will see really soft version of "Dracula". Best, Andrey.

**Comment by albv** [comment]
@dsifford On my Ubuntu 16.04 machine I can't see any difference between themes either. So I took screenshots and used Gimp to compare editor background color - it's the same for both themes, #282a36.

**Comment by dsifford** [comment]
@albv Again, difference can only be seen on macbooks or imacs. For whatever reason, their displays make the default scheme extremely intense.

You are correct that the background color is not touched. That's because it's a non-intense, dark color.

If you look at the build script int this repo, you'll see that only these colors are adjusted:

https://github.com/dracula/visual-studio-code/blob/4975e5a55c5743d097aaafe72ac48b671c994f87/src/dracula.yml#L11-L32
